File: frontend/src/services/authService.js
import api from '../api/axios';

export const authService = {
  login: (credentials) => {
    return api.post('/auth/login', credentials);
  },
  registro: (datos) => {
    return api.post('/auth/registro', datos);
  },
  logout: () => {
    return api.post('/auth/logout');
  },
  me: () => {
    return api.get('/auth/me');
  },
  recuperarPassword: (email) => {
    return api.post('/auth/recuperar-password', { email });
  },
  cambiarPassword: (token, nuevaPassword) => {
    return api.post('/auth/cambiar-password', { token, nuevaPassword });
  },
};

File: backend/src/main/java/com/tiendaonline/service/AuthService.java
package com.tiendaonline.service;

import com.tiendaonline.dto.request.LoginRequest;
import com.tiendaonline.dto.request.RegistroRequest;
import com.tiendaonline.dto.response.AuthResponse;
import com.tiendaonline.dto.response.UserInfoResponse;

public interface AuthService {
    AuthResponse login(LoginRequest request);
    AuthResponse registro(RegistroRequest request);
    AuthResponse refreshToken(String refreshToken);
    UserInfoResponse getCurrentUser(String email);
    void recuperarPassword(String email);
    void cambiarPassword(String token, String nuevaPassword);
}
